text/trace/0107-sampling-priority.md

```diff
@@ -0,0 +1,79 @@
+# Associating sampling priority with the trace using tracestate
+
+Propagating `sampling.priority` (calculated by probability sampler) in a
+`tracestate` will enable a better experience for system with the independent
+stateless configuration of components sampling.
+
+## Motivation
+
+Consistent sampling decision made in each app of a distributed trace is
+important for better user experience of traces analysis. Consistency is achieved
+by aligning stateless hashing algorithm used to make a decision based on
+trace-id or explicitly propagating sampling flag.
+
+Increasingly more apps participate in distributed tracing. With the
+standardized wire format for context propagation, there is bigger chance that
+approach for sampling chosen by one app may not match another app sampling
+policy. Since coordination of sampling policies across many apps not always
+possible, OpenTelemetry must provide a way to share sampling hints between apps.
+
+Beyond sampling hints infrastructure, it is beneficial for the community to
+agree on standard behavior and hints exposed by probability sampler.
+
+The `sampling.priority` property is used by many vendors for various purposes.
+Propagating this field alongside the trace will allow for many improvements and
+as a very minimum will simplify transition of customers from SDKs using
+different `trace-id` based hash functions to OpenTelemetry SDK.
+
+Also see discussion here: <https://github.com/open-telemetry/opentelemetry-specification/pull/570>
+
+## Explanation
+
+### Sampling hints exchange
+
+OpenTelemetry SDK has an infra to expose sampling hints as span attributes today
+via
+[SamplingResult](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#shouldsample)
+class.
+
+This recommendation suggests OpenTelemetry will allow to return an updated
+`tracestate` that will be used for a newly created `Span`. This will allow
+samplers from a single vendor coordinate sampling decisions across the
+different components of the trace.
+
+### Probability Sampler default behavior
+
+The default behavior of probability sampler is to calculate a stateless hash
+function of `trace-id` to make a sampling decision.
+
+It is suggested to extend the sampler behavior to expose a `tracestate` field
+called `sampling.priority` with the integer value `[0-1000]` that will indicate
+sampling priority of the current span.
+
+This will allow to align sampling algorithms between various components.
+Especially for the transition scenarios from the different SDKs.
```

Comment:
Which "transition" scenarios? Switching from an old to a new version? Please clarify.

Reply:
yes. If there are different hashing algos used for historical reasons - you can make compoents respect sampling.priority first and then change algo underneath.

Reply:
it may be expanded in a real spec. @Oberon00 does this note clarify things or you think more descriptive tesxt is needed here. If it's very unclear - @Kanwaldeep worth expanding a bit.

Reply:
Agree, that the discussion in https://github.com/open-telemetry/opentelemetry-specification/pull/611 has people split between using the "random bits" of the UUID from traceid generation, vs. hashing the whole traceid again just in case the trace id was not randomly generated.